elasticache-redis/main.tf

```diff
@@ -80,6 +80,49 @@ resource "aws_kms_key" "ssm_redis_endpoint" {
   tags = local.tags
 }
 
+resource "aws_iam_role" "conduit_ecs_task_role" {
+  name               = "${var.name}-${var.application}-${var.environment}-conduitEcsTask"
+  assume_role_policy = data.aws_iam_policy_document.assume_ecstask_role.json
+
+  inline_policy {
+    name   = "AllowReadingofCMKSecrets"
+    policy = data.aws_iam_policy_document.access_ssm_with_kms.json
+  }
+
+  tags = local.tags
+}
+
+data "aws_iam_policy_document" "assume_ecstask_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
```

Comment:
Can this assume any role? Should we be tying it down to the role it should be able to assume?

Reply:
It should allow the ECS service to assume this role; which is then restricted by the inline policies. The task created by copilot uses dynamic, and unpredictably generated names - so it would be difficult to restrict it any further

Reply:
Fair.